tests/framework/unit_tests/Optimizers/testTournamentSelection.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Testing for tournamentSelection mechanism for parent selection
+  @authors: Mohammad Abdo, Diego Mandelli, Andrea Alfonsi
+"""
+import os
+import sys
+import xarray as xr
+import numpy as np
+
+ravenPath = os.path.abspath(os.path.join(__file__, *['..'] * 5, 'framework'))
+print('... located RAVEN at:', ravenPath)
+sys.path.append(ravenPath)
+import Driver
+from utils import randomUtils
+from Optimizers.parentSelectors.parentSelectors import returnInstance
+
+tournamentSelection = returnInstance('tester', 'tournamentSelection')
+
+#
+#
+# checkers
+#
+def checkSameListOfInt(comment, value, expected, update=True):
+  """
+    This method compares two identical things
+    @ In, comment, string, a comment printed out if it fails
+    @ In, value, list, the value to compare
+    @ In, expected, list, the expected value
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if same
+  """
+  res = not abs(value - expected).any()
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking string", comment, '|', value, "!=", expected)
+      results["fail"] += 1
+  return res
+
+def checkSameDataArrays(comment, resultedDA, expectedDA, update=True):
+  """
+    This method compares two identical things
+    @ In, comment, string, a comment printed out if it fails
+    @ In, resultedDA, xr.DataArray, the resulted DataArray to be tested
+    @ In, expectedDA, xr.DataArray, the expected DataArray
+    @ In, update, bool, optional, if False then don't update results counter
+    @ Out, res, bool, True if same
+  """
+  res = resultedDA.identical(expectedDA)
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking string", comment, '|', resultedDA, "!=", expectedDA)
+      results["fail"] += 1
+  return res
+
+results = {'pass': 0, 'fail': 0}
+
+#
+#
+# formatters
+#
+def formatSample(vars):
+  return dict ((x, np.atleast_1d(y)) for x, y in vars.items())
+
```

Comment:
methods "formatSample", "checkSameDataArrays" and "checkSameListOfInt" appeared in several files, could you make a module to contain these methods instead of duplicate them in several files?